lib/media/streaming_engine.js

```diff
@@ -1325,6 +1325,44 @@ shaka.media.StreamingEngine = class {
 
     mediaState.performingUpdate = true;
 
+    const hanleError = async (error) => {
+      this.destroyer_.ensureNotDestroyed(error);
```

Comment:
So what's the reason for breaking this code into a closure? It's only called once inside fetchAndAppend_, and I don't think that moving the error handling code to the start makes fetchAndAppend_ any easier to read.

Reply:
I reverted this part!